### ISSUE TITLE
微调qwen3-embedding-0.6b模型的时候报错

### ISSUE THREAD

**Issue by zeroTohero4539** [modelscope/ms-swift#5088]
**Describe the bug**
What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)
微调脚本是：
nproc_per_node=8 
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
NPROC_PER_NODE=$nproc_per_node \
swift sft \
    --model /data/zongxiaobing/model/Qwen/Qwen3-Embedding-0.6B \
    --task_type embedding \
    --model_type qwen3_emb \
    --train_type lora \
    --lora_rank 32 \
    --lora_alpha 64 \
    --lora_dropout 0.05 \
    --dataset /data/zongxiaobing/code/cross_language_retrieval/datasets/h01m_crosslingual_for_qwen3_emb_20w.jsonl \
    --split_dataset_ratio 0.05 \
    --eval_strategy steps \
    --output_dir output \
    --eval_steps 50 \
    --num_train_epochs 1 \
    --save_steps 50 \
    --per_device_train_batch_size 2 \
    --per_device_eval_batch_size 2 \
    --gradient_accumulation_steps 2 \
    --learning_rate 3e-4 \
    --loss_type infonce \
    --label_names labels \
    --dataset_num_proc 64 \
    --dataloader_drop_last true \
    --lr_scheduler_type cosine \
    --attn_impl flash_attn \
    --deepspeed zero2
在进行embedding模型微调的时候出现下述错误，每次执行完一个step之后就报错并退出，报错信息如下：

<img width="1194" height="928" alt="Image" src="https://github.com/user-attachments/assets/ae38f0c3-aa8d-406b-a104-b23ca863f389" />

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)
cuda使用的是12.4，系统是ubuntu22.04，GPU是H20*8（96G/卡），torch是2.7.1，数据集有120W条

**Additional context**
Add any other context about the problem here(在这里补充其他信息)

**Comment by tastelikefeet** [comment]
报错不能复现，尝试使用最新的main分支，用一个新的conda环境试试

**Comment by onestardao** [comment]
@zeroTohero4539 

看你的日志和配置，这里更像是两处叠加的问题：

* **No.5 模型/维度不匹配**：训练脚本把 **qwen3-embedding-0.6b** 当作通用 encoder 用了不同的投影或 head，导致 loss/shape 在前几步就爆。先确认 `hidden_size / embedding_dim / pooler` 对齐，评估脚本也别复用 chat 模型的 head。
* **No.11 数据切分与采样**：你用 `pack=True / num_items_in_batch` 一类参数时，拼接超长样本或空样本容易让 dataloader 产出异常 batch（长度为 0 或字段缺失），训练前 5～20 步最容易炸。先固定最大长度，过滤空行，关闭 pack 做最小集跑通。

**最小重现实验**

1. 用 200～500 条干净样本，固定 `max_len`，关掉任何 pack/concat。
2. 直接用官方 qwen3-embedding 模型，按其 `embedding_dim` 建立 linear head，不要复用旧 ckpt。
3. 首轮只跑 50～100 step 验证 loss 下降，再逐步恢复 pack/accumulation。

如果要我给出逐项检查清单（对应问题清单编号），回我「要清单」。我会贴一版可复制的 checklist，你照着对就能把这个训练跑稳。

**Comment by z0o0ey** [comment]
please retry with latest swift and vllm, feel free to reopen or create a new issue if it the problem persists.
